packages/framework-provider-azure-infrastructure/src/infrastructure/rockets/infrastructure-rocket.ts

```diff
@@ -0,0 +1,43 @@
+import { BoosterConfig, RocketDescriptor } from '@boostercloud/framework-types'
+import { FunctionDefinition } from '../types/functionDefinition'
+import { ApplicationSynthStack } from '../types/application-synth-stack'
+import { RocketUtils } from './rocket-utils'
+
+export interface InfrastructureRocket {
+  packageName?: string
+  mountStack: (
+    config: BoosterConfig,
+    applicationSynthStack: ApplicationSynthStack,
+    utils: RocketUtils
+  ) => ApplicationSynthStack
+  unmountStack?: () => void
+  getFunctionAppName?: (applicationSynthStack: ApplicationSynthStack) => string
+  mountFunctions?: (
```

Comment:
What are these for? Are this methods required to be implemented for any rocket?

Reply:
Azure Terraform implementation is able to mount the Azure stack but not to deploy the functions. Right now we are using Azure libraries to do a "zip deploy" with all the FunctionsApp. For the rockets, we need to implement it the same way: stack + functions.

So, those optional methods are only needed if you implement a functionApp in your rocket to allow Booster to do the "Zip deploy" of your function.